vault/resource_azure_secret_backend_role.go

```diff
@@ -150,16 +156,18 @@ func azureSecretBackendRoleUpdateFields(_ context.Context, d *schema.ResourceDat
 		data["azure_groups"] = jsonAzureListString
 	}
 
-	if v, ok := d.GetOk("application_object_id"); ok {
-		data["application_object_id"] = v.(string)
-	}
-
-	if v, ok := d.GetOk("ttl"); ok {
-		data["ttl"] = v.(string)
+	for _, k := range []string{
+		"ttl",
+		"max_ttl",
+		"application_object_id",
+	} {
+		if v, ok := d.GetOk(k); ok {
+			data[k] = v.(string)
+		}
 	}
 
-	if v, ok := d.GetOk("max_ttl"); ok {
-		data["max_ttl"] = v.(string)
+	if v, ok := d.GetOk("permanently_delete"); ok {
```

Comment:
We should use `d.Get` for booleans otherwise this will cause bugs. See https://discuss.hashicorp.com/t/terraform-sdk-usage-which-out-of-get-getok-getokexists-with-boolean/41815

Reply:
Good catch! Thank you!